infra/cifuzz/fuzz_target.py

```diff
@@ -232,17 +250,55 @@ def download_oss_fuzz_build(self):
     if not latest_build_str:
       return None
 
-    oss_fuzz_build_url = url_join(GCS_BASE_URL, self.project_name,
-                                  latest_build_str)
-    try:
-      urllib.request.urlretrieve(oss_fuzz_build_url, BUILD_ARCHIVE_NAME)
-    except urllib.error.HTTPError:
-      logging.error('Unable to download build from: %s.', oss_fuzz_build_url)
+    oss_fuzz_build_url = url_join(GCS_BASE_URL, CLUSTERFUZZ_BUILDS,
+                                  self.project_name, latest_build_str)
+    return download_zip(oss_fuzz_build_url, build_dir)
+
+  def download_latest_corpus(self):
+    """Downloads the latest OSS-Fuzz corpus for the target from google cloud.
+
+    Returns:
+      The local path to to corpus or None if download failed.
+    """
+    if not os.path.exists(self.out_dir):
+      logging.error('Out directory %s does not exist.', self.out_dir)
       return None
-    with zipfile.ZipFile(BUILD_ARCHIVE_NAME, 'r') as zip_file:
-      zip_file.extractall(build_dir)
-    os.remove(BUILD_ARCHIVE_NAME)
-    return build_dir
+    if not self.project_name:
+      return None
+
+    corpus_dir = os.path.join(self.out_dir, 'backup_corpus', self.target_name)
+    os.makedirs(corpus_dir, exist_ok=True)
+    corpus_url = url_join(
+        GCS_BASE_URL,
+        '{0}-backup.clusterfuzz-external.appspot.com/corpus/libFuzzer/'.format(
+            self.project_name),
+        '{0}_{1}'.format(self.project_name, self.target_name), CORPUS_ZIP_NAME)
+    return download_zip(corpus_url, corpus_dir)
+
+
+def download_zip(http_url, out_dir):
+  """Downloads a zip file from an http url.
+
+  Args:
+    http_url: A url to the zip file to be downloaded.
+    out_dir: The path where the zip file should be extracted to.
+
+  Returns:
+    A path to the extracted file or None on failure.
+  """
+  if not os.path.exists(out_dir):
+    logging.error('Out directory %s does not exist.', out_dir)
+    return None
+  tmp_file = os.path.join(out_dir, 'tmp.zip')
```

Comment:
can this code ever run in parallel on the same image? If yes, it might make sense to use a non constant archive name, or at least make it depend on the fuzz target name, e.g. `<fuzz_target_>_public_corpus_backup.zip`

Reply:
There are no plans to make it parallel but having the option would be good. I will update it. Does the way I implemented it make sense to you?